### PR TITLE
Add playlist dropdown to Playstyle

### PR DESCRIPTION
**Pull request by Sciguymjm**
First real React PR. Please review.

The dropdown selection display is broken and I'm not sure how to fix it. But selecting a playlist updates the graphs.

### CODE REVIEW

File: webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyle.tsx [hunk 1]
```diff
@@ -73,7 +79,7 @@ class PlayerPlayStyleComponent extends React.PureComponent<Props, State> {
     }
 
     private readonly getPlayStyles = (): Promise<void> => {
-        return getPlayStyle(this.props.player.id)
+        return getPlayStyle(this.props.player.id, undefined, this.props.playlist)
```
Comment (Sciguymjm) [inline webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyle.tsx]: I'm not sure how to skip the rank parameter and just use playlist, so I had to use `undefined` here.

File: webapp/src/Requests/Player/getPlayStyle.ts [hunk 2]
```diff
@@ -38,5 +39,6 @@ export const getPlayStyle = (id: string, rank?: number): Promise<PlayStyleRespon
             }
         )
     }
-    return doGet(`/player/${id}/play_style` + (rank === undefined ? "" : `?rank=${rank}`))
+    const url = qs.stringify({rank, playlist}, {addQueryPrefix: true, indices: false})
```
Comment (Sciguymjm) [inline webapp/src/Requests/Player/getPlayStyle.ts]: I use stringify here and it seems to work nicely.

File: webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyleCard.tsx [hunk 3]
```diff
@@ -20,7 +21,9 @@ export class PlayerPlayStyleCard extends React.PureComponent<Props> {
 
         return (
             <Card>
-                <CardHeader title={playStyleTitle} action={<PlayStyleActions player={this.props.player}/>}/>
+                <CardHeader title={playStyleTitle} action={<PlayStyleActions
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyleCard.tsx]: put action on the next line - the current whitespacing makes "player" and "handleplaylistchange" look like props of CardHeader.

File: webapp/src/Components/Player/PlayerOverview.tsx [hunk 4]
```diff
@@ -99,6 +103,11 @@ class PlayerOverviewComponent extends React.PureComponent<Props, State> {
     private readonly handleSelectMobileTab = (event: React.ChangeEvent, selectedMobileTab: PlayerViewTab) => {
         this.setState({selectedMobileTab})
     }
+
+    private readonly handlePlaylistChange = (playlist: number) => {
+        console.log("Got playlist change in Page", playlist)
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/PlayerOverview.tsx]: pls remove. (along with the other console logs)

File: webapp/src/Components/Player/PlayerOverview.tsx [hunk 5]
```diff
@@ -22,12 +22,13 @@ const playerViewTabs = ["Profile", "Playstyle", "Match History"]
 
 interface State {
     selectedMobileTab?: PlayerViewTab
+    playlist?: number
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/PlayerOverview.tsx]: why optional?

File: webapp/src/Components/ReplaysSearch/Filter/ReplaysSearchFilter.tsx [hunk 6]
```diff
@@ -32,7 +32,9 @@ export class ReplaysSearchFilter extends React.PureComponent<Props> {
             <PlaylistSelect selectedPlaylists={queryParams.playlists || []}
                             handleChange={this.handlePlaylistsChange}
                             inputLabel="Playlist"
-                            helperText="Select playlist to filter by"/>
+                            helperText="Select playlist to filter by"
+                            multiple={true}
```
Comment (twobackfromtheend) [inline webapp/src/Components/ReplaysSearch/Filter/ReplaysSearchFilter.tsx]: `prop={true}`  can be shortened as `prop`.

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx [hunk 7]
```diff
@@ -20,6 +20,7 @@ interface PlaylistMetadata {
     current: boolean
     ranked: boolean
     standardMode: boolean
+    publicMode: boolean
```
Comment (twobackfromtheend) [inline webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: Do you know of a better place to put this?

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx [hunk 8]
```diff
@@ -170,6 +191,9 @@ interface OwnProps {
     handleChange: React.ChangeEventHandler<HTMLSelectElement>
     helperText: string
     inputLabel: string
+    justDropdown?: boolean
```
Comment (twobackfromtheend) [inline webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: `dropdownOnly` and `currentPlaylistsOnly` are probably more standard names.

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx [hunk 9]
```diff
@@ -179,16 +203,22 @@ interface State {
...
     filterRanked: boolean
     filterStandardMode: boolean
+    filterPublic: boolean // Just playlists that exist on play online menu, in which we can get consistent data
     optionsExpanded: boolean
 }
 
 class PlaylistSelectComponent extends React.PureComponent<Props, State> {
     constructor(props: Props) {
         super(props)
+        let shouldFilterPublic = false
```
Comment (twobackfromtheend) [inline webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: `const shouldFilterPublic = props.currentPlaylistsOnly ? true : false`

or `const ... = Boolean(props.cakslaskdlaskd)`.

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx [hunk 10]
```diff
@@ -272,35 +303,45 @@ class PlaylistSelectComponent extends React.PureComponent<Props, State> {
...
-                    {playlistsMultiSelect}
-                </ExpansionPanelSummary>
-                <ExpansionPanelDetails className={classes.panelDetails}>
-                    {filterCurrentCheckbox}
-                    {filterStandardCheckbox}
-                    {filterRankedCheckbox}
-                </ExpansionPanelDetails>
-            </ExpansionPanel>
+        return (<>
+
```
Comment (twobackfromtheend) [inline webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: put the `<>` on line 307.

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx [hunk 11]
```diff
@@ -272,35 +303,45 @@ class PlaylistSelectComponent extends React.PureComponent<Props, State> {
...
+                    <ExpansionPanelDetails className={classes.panelDetails}>
+                        {filterCurrentCheckbox}
+                        {filterStandardCheckbox}
+                        {filterRankedCheckbox}
+                    </ExpansionPanelDetails>
+                </ExpansionPanel>
+
+                }
+                {
+                    this.props.justDropdown && <>{playlistsMultiSelect}</>
```
Comment (twobackfromtheend) [inline webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: Delete the blank line above, and collapse this bracket to a single line. `<>{playlistsMultiSelect}</>` should be shortenable to `playlistsMultiSelect`.

File: webapp/src/Components/Player/PlayerOverview.tsx (hunk 5 above)
Comment (Sciguymjm) [reply webapp/src/Components/Player/PlayerOverview.tsx]: Its optional on the server.

File: webapp/src/Components/Shared/Selects/PlaylistSelect.tsx (hunk 7 above)
Comment (Sciguymjm) [reply webapp/src/Components/Shared/Selects/PlaylistSelect.tsx]: Its another filtering option IMO.

File: webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyle.tsx (hunk 1 above)
Comment (Abbondanzo) [reply webapp/src/Components/Player/Overview/PlayStyle/PlayerPlayStyle.tsx]: This is standard: https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

**Pull request merged**